ArgoProject/src/systems/CollisionSystem.cpp

```diff
@@ -0,0 +1,81 @@
+#include "stdafx.h"
+#include "CollisionSystem.h"
+
+// components
+#include "components/Location.h"
+#include "components/Dimensions.h"
+#include "components/Render.h"
+#include "components/Collision.h"
+#include "components/Input.h"
+#include "components/Motion.h"
+
+app::sys::CollisionSystem::CollisionSystem()
+	: BaseSystem()
+{
+	//prepare these components
+	m_registry.prepare<comp::Collision, comp::Location, comp::Input, comp::Dimensions>();
+}
+
+void app::sys::CollisionSystem::update(app::time::seconds const & dt)
+{
+	//type of collision
+	std::string manifoldType;
+
+	//view player
+	m_registry.view<comp::Collision, comp::Input, comp::Location, comp::Dimensions>(entt::persistent_t())
+		.each([&, this](app::Entity const entity, comp::Collision & collision, comp::Input & input, comp::Location & location, comp::Dimensions & dimensions)
+	{
+		//update collision boxes
+		updateAABB(std::get<cute::c2AABB>(collision.collisionBox), location.position, dimensions.size, dimensions.origin);
```

Comment:
std::variant best practice use `std::visit` function in which you pass in a object that has a overloaded member function `operator()` that accepts all the types that the variant contains.
```suggestion
		std::visit(vis::CollisionTrackingVisitor(location, dimensions), collision.collisionBox);
```
in namespace `app::vis`
```cpp
class CollisionTrackingVisitor
{
public:
		CollisionTrackingVisitor(comp::Location const & location, comp::Dimensions const & dimensions);
		operator()(cute::c2AABB & aabb) const
		{
				aabb.min = cute::c2v{
						m_location.position.x - m_dimensions.origin.x,
						m_location.position.y - m_dimensions.origin.y
				};
				aabb.max = cute::c2v{
						aabb.min.x + m_dimensions.size.x,
						aabb.min.y + m_dimensions.size.y
				};
		}
		operator()(cute::c2Circle & circle) const
		{
				circle.p.x = m_location.position.x; circle.p.y = m_location.position.y;
				circle.r = m_dimensions.size.x > m_dimensions.size.y ? m_dimensions.size.x : m_dimensions.size.y;
		}
private:
		comp::Location const & m_location;
		comp::Dimensions const & m_dimensions;
};
```
Note: for simplicity I have implementation in the class declaration when it should be in the `.cpp`